pkg/controller/registry/resolver/cache.go

```diff
@@ -151,11 +158,20 @@ func (c *OperatorCache) Namespaced(namespaces ...string) MultiCatalogOperatorFin
 
 	for _, miss := range misses {
 		ctx, cancel := context.WithTimeout(context.Background(), CachePopulateTimeout)
+
+		catsrcPriority := 0
+		// Ignoring error and treating catalog source priority as 0 if not found
+		catsrc, err := c.crClient.OperatorsV1alpha1().CatalogSources(miss.Namespace).Get(ctx, miss.Name, v1.GetOptions{})
```

Comment:
Is it relatively safe to get this value from the cache instead of hitting the API server directly?

Reply:
we don't have the value in cache yet, this is the first time getting it. Once we do, they are be resued until snapshot expires.

Reply:
I'm referring to the client-go cache of CatalogSources here, not the resolver's operator cache. Shouldn't the catalog operator always have a CatalogSource cached by the time it's needed here? Making a real request to the API server may be unnecessary.

Reply:
good call, taking a look

Reply:
https://github.com/operator-framework/operator-lifecycle-manager/pull/1688/commits/6c2f414c727a3f98804adcea255ef2876b91561e

Reply:
git cherry-picked those cache changes.